Draft: Added section recommended literature to README.md

Enough7 opened the pull request:
I will probably add more soon.
Is the `README.md` the right place for this?
Maybe you want such information in a different place such as Artery's website.

# TODO
- [x] Find resolution regarding broken images

riebl commented:
I like this idea! As you have already announced that there will be more, adding it to a page on the Artery website might be better indeed. We could also add a subsection with articles using Artery there. In the `README.md` we can add a link to the webpage.
Do you have experience with *mkdocs* which is the current website generator for Artery?

Enough7 commented:
> Do you have experience with mkdocs which is the current website generator for Artery?

I have experience with MarkDown.
The website's repository is private, am I right?
In general I have some experience in NodeJS and React, mainly using TypeScript.

> As you have already announced that there will be more.

There will be more given that I will find more.

awillecke commented:
> I have experience with MarkDown. The website's repository is private, am I right? In general I have some experience in NodeJS and React, mainly using TypeScript.

I guess the website's sources are located in [docs](https://github.com/riebl/artery/tree/master/docs).

Enough7 commented:
> I guess the website's sources are located in [docs](https://github.com/riebl/artery/tree/master/docs).

I see, I see, thank you

Enough7 commented:
I added on how to build the Website and added a `pyproject.toml` for easy install.
Also I can not syntaxhighlighting using  `codehilite` with `pygments` to run.
Any ideas on this?

Enough7 commented:
Fixed path issue.
Also the `docs/asset/vagrant.webm` has for some reason no header.
I am not sure if I broke it or it broke somehow else.

Enough7 commented:
The Doxygen Docs could also be integrated into the website, don't you think?

Enough7 commented:
> Also the `docs/asset/vagrant.webm` has for some reason no header. I am not sure if I broke it or it broke somehow else.

The `logo.png` seem broken as well, for some reason?
![image](https://github.com/riebl/artery/assets/30051680/1759c13f-2956-462f-96b3-b914d2702712)

Enough7 commented:
# Request for feedback
Optional darkmode (uses User System Settings, can be switch manually in the header):
![Optional darkmode ](https://github.com/riebl/artery/assets/30051680/cc726c0f-4a25-4f9f-9162-daf911d3b32c)

Added Syntaxhighlighting and line numbers:
![image](https://github.com/riebl/artery/assets/30051680/dba2ba0c-d99c-4fe8-aee5-a9ab31447a8f)

Also made some changes to 'Work with VS Code' page.

I suggest you to start at the [`docs/website/README.md`](https://github.com/Enough7/artery/blob/ReadMeLiterature/docs/website/README.md),

Enough7 commented:
Any ideas on why `theme.palette` only works with two spaces indention and does not work with four spaces indention?
See [`mkdocs.yml` Line 9-31)](https://github.com/Enough7/artery/blob/ReadMeLiterature/docs/mkdocs.yml#L9-L31)

Enough7 commented:
Next step could be
- [ ] to use [abbreviations and  glossaries](https://squidfunk.github.io/mkdocs-material/reference/tooltips/?h=glossary#adding-abbreviations)
- [ ] figure out a styleguide
    - e.g. use mono or italic for code, but not both
- [ ] add codedocs somewhere (e.g. generated by doxygen)

riebl commented:
Oops, I have added review comments but forgot to submit my review a few days ago…

Enough7 commented on docs/assets/vagrant.webm:
[It somehow coruppted](https://github.com/riebl/artery/pull/301#issuecomment-1652366438) so I downloaded it from the website and replaced it.

riebl replied:
That is very odd: I have just downloaded the video file from the website and put it into *docs/assets*. No changes at all.

Enough7 replied:
That's weird. That's the first time that this happened to me. 

> Edit

But how is it possible that whole segments in that file are missing?

Enough7 replied:
I will revert it by rebasing, when I clean up the git history before the pull eventually.

Enough7 replied:
Should be fixed with [24f0352ada483f988c590f1bed608b432c3b5c0f](https://github.com/riebl/artery/pull/301/commits/24f0352ada483f988c590f1bed608b432c3b5c0f#diff-771e96f3b03a66c57dede75acf59435d151ea5ffd60a8b31e264cdb2433b5794)

Enough7 commented on docs/mkdocs.yml:
I guess it is more an opinion than a fact but I think it is beneficial to have most files grouped in directories and only few files floating around in the root directory.
It is more structured and so developers can faster recognize the structure.
I moved because I think its an subproject and therefore should be in dedicated subdirectory.

riebl replied:
Okay, fair point.

Enough7 commented on docs/mkdocs.yml:
```diff
@@ -1,31 +1,49 @@
+        scheme: slate
         primary: 'teal'
         accent: 'deep orange'
-    font:
-        text: 'Lato'
-        code: 'Oxygen Mono'
```
TODO: I messed this up

Enough7 replied:
Fixed in [0dd85cb8350b36b1aa7d1727c02fc345fae56a00](https://github.com/riebl/artery/pull/301/commits/0dd85cb8350b36b1aa7d1727c02fc345fae56a00)

Enough7 commented on docs/website/vscode.md:
I did this because of possible future changes.
For example the Doxygen Docs would be placed in `docs/doxygen`

riebl replied:
I don't like the additional *website* layer because the markdown files are not generated. However, I am fine with subdirectories for generated content such as *docs/doxygen*.

Enough7 replied:
So should I revert it?

Enough7 replied:
If I remember correctly, I think I did this as I said in the other comment section, to give more structure to the directory tree. I wanted to move the `WWW`-build directory to `docs/` as well, but I thought what if riebl has some automated deploy script. That would be breaking that script if it does expect that directory to be in the root directory. I forget to ask you if I should move `WWW` to `docs/`. 
Should I move it?

Then we got three subdirectories in `docs/`.

riebl replied:
Generating the website at *docs/www* is fine. I have a script here to upload new website versions, but that is easy to adjust.

Enough7 replied:
Move the folder to `docs/www` in [b34bde16e02d56426f3befceec01a9c86d5364ca](https://github.com/riebl/artery/pull/301/commits/b34bde16e02d56426f3befceec01a9c86d5364ca)

Enough7 commented on docs/install.md:
```diff
@@ -130,7 +130,9 @@ You can run the example from the *artery-build* directory located in the box's h
 invoking `make run_example`.
 Please note that the *artery* directory mirrors the cloned repository on your host machine.
 
-![Starting Artery in virtual machine supplied by Vagrant](../assets/vagrant.webm){:autoplay loop}
```
I have noticed it but I could not get a working version of vidifiy.

> I think mixing two mark-up languages in a single file is odd.

I agree except I think of Markdown more as a preprocessing language for html. So you just take away some work from the compile :D.
Also I think it is beneficial to have as little as possible dependencies.

Enough7 replied:
Maybe you can add vidifiy to the [dependencies](https://github.com/riebl/artery/pull/301/files#diff-001ee8fd3c7ce6043e2474c96960d83c5a0eff948e799f27763e996b283c4dc6R9)?

riebl replied:
*pip install markdown-vidify* did the trick on my system. Can you please elaborate on the problems you are facing with it?

I am more often reading the raw markdown files than the generated website. That's probably why I want the markdown to stay tidy :-)

Enough7 replied:
> Can you please elaborate

Back than I tried to install it with `apt` or `pip install vidify` did not work.
I have to say that I usually kept away from the python ecosystem.

> promising alternative

I think it would be wrong to use such plugins but I really do not care.
From my experience I think usually the more third party code you got the harder to maintain and the easier to penetrate the application is.
From my perspective if I see something like this `![type:video](video.mp4)` I would be confused.
If I see some unknown HTML-Element I know where I can look it up.
I see Markdown as a shortform of HTML.
I do not think that is really that bad.

I will check if I can simply revert that commit.

Enough7 replied:
Vidify works for me now and I reverted that commit.

Enough7 commented on docs/.vscode/settings.json:
Would you mind a `docs/.vscode` or `docs/.vscode/settings.json.example`?
I think YAML schema validation for `mkdocs.yml` is quite a powerful support while coding, so I would have been happy if it was given to me beforehand.

riebl replied:
I would not mind something like *docs/vscode-mkdocs-settings.json*.
As a rule of thumb: Don't interfere with users' already existing settings and avoid hidden files where possible.

Enough7 replied:
Renamed it in [7886591339c685d16fa10a416faad4a846888b8d](https://github.com/riebl/artery/pull/301/commits/7886591339c685d16fa10a416faad4a846888b8d)

riebl commented on docs/mkdocs.yml:
```diff
@@ -1,31 +1,49 @@
+        toggle:
+          icon: material/weather-night
+          name: Switch to dark mode
+      # Palette toggle for dark mode
```
Yes, I just don't understand the fuss about having a darkmode.

Enough7 replied:
It saves energy which is useful in mobile applications especially.

riebl commented:
Hi @Enough7, I picked most of your commits from this draft PR into the master branch.
I think I have only omitted the VSCode YAML schema for now. Please feel free to put this schema file in a new PR; a short info about this file and its purpose in *docs/mkdocs.yml* would be nice too!
Thanks a lot for your hard work so far!

PS: I cannot merge PRs that are marked as "draft", thus I had to pick your commits locally.

Enough7 commented:
Hi @riebl, I just saw the homepage and it looks really great I think.

> I cannot merge PRs that are marked as "draft", thus I had to pick your commits locally. 

Ohh thank you for letting me know. I did not know that. In GitLab it is usually the prefix `Draft:` which needs to be removed from the tittle in order to merge. I am more used to GitLab than to GitHub

Enough7 commented:
Hi @riebl,
I just want to add the following.
- I would recommend you to rename `docs/mkdocs.md` to `README.md` because otherwise it is not shown in the Folders preview and hard to find
- Also I would not use Pythons `requirements.txt` because it usually saves indirect dependencies as well. These could be comitted to the repository,
